### PR TITLE
Udmi present value

### PR DESCRIPTION
**Pull request by sypks**

### CODE REVIEW

File: tools/validators/instance_validator/tests/entity_instance_test.py [hunk 1]
```diff
@@ -97,9 +97,9 @@ def testCombinedChecksInstanceAndGraph(self, mock_entity, mock_gv, mock_iv):
     validator = entity_instance.CombinationValidator(self.config_universe,
                                                      _UPDATE_CFG, {})
 
-    self.assertTrue(validator.Validate(mock_entity))
-    mock_iv.assert_called_once_with(mock_entity)
-    mock_gv.assert_called_once_with(mock_entity)
+    self.assertTrue(validator.Validate(mock_entity()))
+    mock_iv.assert_called_once_with(mock_entity.return_value, False)
+    mock_gv.assert_called_once_with(mock_entity.return_value)
```
Comment (charbull) [inline tools/validators/instance_validator/tests/entity_instance_test.py:102]: with true/false?

File: tools/validators/instance_validator/tests/entity_instance_test.py [hunk 2]
```diff
@@ -1039,5 +1039,39 @@ def testValidateGoodUpdateOperationDefaultExport(self):
...
+    entity_instances = {}
+    for entity_guid, entity_parsed in parsed.items():
+      entity = entity_instance.EntityInstance.FromYaml(
+          entity_guid, entity_parsed, default_operation=default_operation)
+      entity_instances[entity_guid] = entity
+    combination_validator = entity_instance.CombinationValidator(
+        self.config_universe, _INIT_CFG, entity_instances)
+
+    self.assertFalse(combination_validator.Validate(
+      entity_instances['SDC_EXT-17-GUID'], True))
```
Comment (charbull) [inline tools/validators/instance_validator/tests/entity_instance_test.py:1057]: nit: `is_udmi=True`

File: tools/validators/instance_validator/tests/entity_instance_test.py [hunk 3]
```diff
@@ -1039,5 +1039,39 @@ def testValidateGoodUpdateOperationDefaultExport(self):
...
+    entity_instances = {}
+    for entity_guid, entity_parsed in parsed.items():
+      entity = entity_instance.EntityInstance.FromYaml(
+          entity_guid, entity_parsed, default_operation=default_operation)
+      entity_instances[entity_guid] = entity
+    combination_validator = entity_instance.CombinationValidator(
+        self.config_universe, _INIT_CFG, entity_instances)
+
+    self.assertTrue(combination_validator.Validate(
+      entity_instances['SDC_EXT-17-GUID'], True))
```
Comment (charbull) [inline tools/validators/instance_validator/tests/entity_instance_test.py:1074]: same here please.

File: tools/validators/instance_validator/tests/fake_instances/BAD/translation_udmi_present_value.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 Google LLC
```
Comment (charbull) [inline tools/validators/instance_validator/tests/fake_instances/BAD/translation_udmi_present_value.yaml:1]: 2022

File: tools/validators/instance_validator/validate/entity_instance.py [hunk 5]
```diff
@@ -258,14 +264,18 @@ def _ValidateType(self, entity: EntityInstance) -> bool:
...
+    entity: EntityInstance,
+    is_udmi: bool= False) -> bool:
     """Validate an entity's translation against the entity's type or ontology.
 
     If entity operation is ADD, this code ensures that all fields are in the
     defined type and all required fields of the type are defined.
 
     Args:
       entity: EntityInstance to validate
+      is_udmi: flag to validate under udmi
```
Comment (charbull) [inline tools/validators/instance_validator/validate/entity_instance.py:278]: add: `defaults to false` ?

File: tools/validators/instance_validator/validate/entity_instance.py [hunk 6]
```diff
@@ -499,11 +514,12 @@ def _LinkStatesMatch(self, source_field: str, target_field: str) -> bool:
       return False
     return True
 
-  def Validate(self, entity: EntityInstance) -> bool:
+  def Validate(self, entity: EntityInstance, is_udmi: bool= False) -> bool:
     """Uses the generated ontology universe to validate an entity.
 
     Args:
       entity: EntityInstance to validate
+      is_udmi: flag to validate under udmi
```
Comment (charbull) [inline tools/validators/instance_validator/validate/entity_instance.py:522]: add: defaults to false ?

**Review by charbull — CHANGES_REQUESTED** [review]
thank you Shane !
minor changes

File: tools/validators/instance_validator/validate/entity_instance.py (hunk 6 above)
Comment (sypks) [reply tools/validators/instance_validator/validate/entity_instance.py]: done

File: tools/validators/instance_validator/tests/fake_instances/BAD/translation_udmi_present_value.yaml (hunk 4 above)
Comment (sypks) [reply tools/validators/instance_validator/tests/fake_instances/BAD/translation_udmi_present_value.yaml]: fixed

File: tools/validators/instance_validator/tests/entity_instance_test.py (hunk 3 above)
Comment (sypks) [reply tools/validators/instance_validator/tests/entity_instance_test.py]: np!

File: tools/validators/instance_validator/tests/entity_instance_test.py (hunk 2 above)
Comment (sypks) [reply tools/validators/instance_validator/tests/entity_instance_test.py]: ty!

File: tools/validators/instance_validator/tests/entity_instance_test.py (hunk 1 above)
Comment (sypks) [reply tools/validators/instance_validator/tests/entity_instance_test.py:105]: we understood this as ensuring that we called the mock with False and not True for the udmi flag.
the change ensures it was called with is_udmi= False

File: .github/workflows/python_tools.yml [hunk 7]
```diff
@@ -15,7 +15,6 @@ on:
 
 jobs:
   build:
-
```
[the remark this answers is not in the document]
Comment (sypks) [reply .github/workflows/python_tools.yml]: apologies, thank you!

**Comment by sypks** [comment]
changes made :)

File: tools/validators/instance_validator/validate/handler.py [hunk 8]
```diff
@@ -219,13 +220,15 @@ def __init__(self, universe: pvt.ConfigUniverse):
...
-      config_mode: instance_parser.ConfigMode
+      config_mode: instance_parser.ConfigMode,
+      is_udmi: bool= False
   ) -> Dict[str, entity_instance.EntityInstance]:
     """Validates entity instances that are already deserialized.
 
     Args:
       entities: a dict of entity instances
       config_mode: processing mode of the configuration
+      is_udmi: flag to indicate validation under udmi
```
Comment (charbull) [inline tools/validators/instance_validator/validate/handler.py:231]: defaults to false?

**Review by charbull — CHANGES_REQUESTED** [review]
minor

**Comment by sypks** [comment]
verified working telemetry validation locally.
timeout bug discovered

File: tools/validators/instance_validator/validate/entity_instance.py [hunk 9]
```diff
@@ -112,13 +118,13 @@ def __init__(self, universe: pvt.ConfigUniverse,
     self.config_mode = config_mode
     self.entity_instances = entity_instances
 
-  def Validate(self, entity: EntityInstance) -> bool:
+  def Validate(self, entity: EntityInstance, is_udmi: bool= False) -> bool:
     """Returns true if an entity follows all instance and graph rules."""
```
Comment (charbull) [inline tools/validators/instance_validator/validate/entity_instance.py:122]: can you add docstring please

**Review by charbull — APPROVED** [review]

File: tools/validators/instance_validator/validate/entity_instance.py (hunk 9 above)
Comment (sypks) [reply tools/validators/instance_validator/validate/entity_instance.py]: done!

**Pull request merged**